### ISSUE TITLE
ImageNet example for single image classification

### ISSUE THREAD

**Issue by FuriouslyCurious** [pytorch/examples#134]
Hi Soumith,

Do you have an example for single image classification code?  I am trying to load a checkpointed model and classify a single image with this code, but I get a "3D tensor expected" error.

Code:
```
# Bunch of imports go here

# Convert image to Variable
def Torchify( aImage ):
    ptLoader = transforms.Compose([transforms.ToTensor()])
    aImage = ptLoader( aImage ).float()
    aImage = Variable( aImage, volatile=True  )
    return aImage.cuda()

# Load model from Checkpoint
print("=> Loading Network")
ptModelAxial = densenet.__dict__['densenet161'](pretrained=False, num_classes=5)
ptModelAxial.classifier = nn.Linear(8832, 5)
ptModelAxial = torch.nn.DataParallel(ptModelAxial).cuda()
dTemp = torch.load("best.pth.tar")
ptModelAxial.load_state_dict(dTemp['state_dict'])
for p in ptModelAxial.parameters():
    p.requires_grad = False
ptModelAxial.eval()

InputImg = skimage.img_as_float(skimage.io.imread(sFileName))
ptModelPreds = ptModelAxial( Torchify(InputImg) )
print( ptModelPreds )
```

Error message:

```
Traceback (most recent call last):
  File "extract.py", line 298
      ptModelPreds = ptModelAxial( Torchify(InputImg) )
File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/modules/module.py", line 206, in __call__
    result = self.forward(*input, **kwargs)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/parallel/data_parallel.py", line 61, in forward
    outputs = self.parallel_apply(replicas, inputs, kwargs)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/parallel/data_parallel.py", line 71, in parallel_apply
    return parallel_apply(replicas, inputs, kwargs)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/parallel/parallel_apply.py", line 45, in parallel_apply
    raise output
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/parallel/parallel_apply.py", line 25, in _worker
    output = module(*input, **kwargs)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/modules/module.py", line 206, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/keyur/kaggle/densenet.py", line 153, in forward
    features = self.features(x)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/modules/module.py", line 206, in __call__
    result = self.forward(*input, **kwargs)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/modules/container.py", line 64, in forward
    input = module(input)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/modules/module.py", line 206, in __call__
    result = self.forward(*input, **kwargs)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/modules/conv.py", line 237, in forward
    self.padding, self.dilation, self.groups)
  File "/conda3/envs/idp/lib/python3.5/site-packages/torch/nn/functional.py", line 39, in conv2d
    return f(input, weight, bias)
RuntimeError: expected 3D tensor
```

**Comment by Atcold** [comment]
@FuriouslyCurious, would you mind posting the output of `print(InputImg.shape)`?

**Comment by fehiepsi** [comment]
@FuriouslyCurious There is a thing to keep in mind when dealing with images in PyTorch: you have to feed into the model Tensor of size `batch_size x num_channels x height x width`. Assume that your image has a correct order of `num_channels x height x width`, use have to do e.g.  `aImage = aImage.unsqueeze(0)` to add an additional dimension for minibatch (of course, with size 1), your tensor now is in the size `1 x num_channels x height x width`, which is good to feed into the model.

**Comment by varun-suresh** [comment]
I had a similar error. I checked my input shape and it was `1x3x112x96`. It threw the `Runtimeerror: expected 3D tensor` right before my fully connected layer. When I changed my batch size to 2, input shape was `2x3x112x96` it worked without any errors. 

In my case, the `squeeze` before the fully connected layer was causing the error. When the batch size was 1, the output shape after the squeeze was `[132]` instead of `[1,132]`. Once I corrected that, it worked fine.
